Services/EmployeeShiftService.cs

```diff
@@ -6,48 +6,64 @@ namespace conscoord_api.Services;
 
 public class EmployeeShiftService : IEmployeeShiftService
 {
-    readonly PostgresContext _context;
-    public EmployeeShiftService(PostgresContext context)
-    {
-        _context = context;
-    }
+  readonly PostgresContext _context;
+  public EmployeeShiftService(PostgresContext context)
+  {
+    _context = context;
+  }
 
-    public async Task CreateEmployeeShift(EmployeeShift empShift)
-    {
-        _context.EmployeeShifts.Add(empShift);
-        await _context.SaveChangesAsync();
-    }
+  public async Task CreateEmployeeShift(EmployeeShift empShift)
+  {
+    _context.EmployeeShifts.Add(empShift);
+    await _context.SaveChangesAsync();
+  }
 
-    public List<Shift> GetScheduledShiftsByEmpId(int empId)
-    {
-        return _context.Shifts
-            .Where(s => s.EmployeeShifts.Any(es => es.EmpId == empId))
-            .ToList();
-    }
+  public List<Shift> GetScheduledShiftsByEmpId(int empId)
+  {
+    return _context.Shifts
+        .Where(s => s.EmployeeShifts.Any(es => es.EmpId == empId))
+        .ToList();
+  }
 
-    public async Task DeleteEmpShiftAsync(int empShiftId)
-    {
-        EmployeeShift? shift = _context.EmployeeShifts
-            .Where(s => s.Id == empShiftId)
-            .FirstOrDefault();
+  public async Task DeleteEmpShiftAsync(int empShiftId)
+  {
+    EmployeeShift? shift = _context.EmployeeShifts
+        .Where(s => s.Id == empShiftId)
+        .FirstOrDefault();
 
-        if (shift != null)
-        {
-            _context.EmployeeShifts.Remove(shift);
-            await _context.SaveChangesAsync();
-        }
+    if (shift != null)
+    {
+      _context.EmployeeShifts.Remove(shift);
+      await _context.SaveChangesAsync();
     }
+  }
 
   public List<EmployeeShift> GetFutureShifts()
   {
     DateTime currentTime = DateTime.Now;
     var futureShifts = _context.EmployeeShifts
       .Include(s => s.Shift)
       .Include(e => e.Emp)
-      .AsEnumerable() // Bring the data into memory
     .Where(s => DateTime.ParseExact(s.Shift.StartTime, "yyyy/MM/dd HH:mm:ss", null) >= currentTime)
     .ToList();
 
     return futureShifts;
   }
+
+  public List<EmployeeShift> GetShiftsWithinTime(DateTime start, DateTime end)
+  {
+    var shifts = _context.EmployeeShifts
+        .Include(s => s.Shift)
+        .Include(e => e.Emp)
+        .AsEnumerable() // Switch to in-memory processing
+        .Where(s =>
+        {
+          DateTime startTime;
+          bool parsed = DateTime.TryParseExact(s.Shift.StartTime, "yyyy/MM/dd HH:mm:ss", null, System.Globalization.DateTimeStyles.None, out startTime);
+          return parsed && startTime >= start && startTime <= end;
+        })
+        .ToList();
+
+    return shifts;
+  }
```

Comment:
We should have tests for this + GetFutureShifts(). In the meantime, we can also bring in the other shift we forgot to bring in from the practicum2425 project.